src/peak_finding/1d_list.clj

```diff
@@ -0,0 +1,82 @@
+;; A peak is any element that is larger or equal to both it's neighbours
+;; Items that only have one neighbour (end items) are considered a peak if
+;; they are larger or equal to their neighbour
+;; This algorithm returns the first peak it finds.
+
+(ns peakfinder)
+
+(defn ^:dynamic find-peak
```

Comment:
There is really no reason to make this ^:dynamic.  You really only need dynamic if you intend to rebind it dynamically in a (rebind [...]) form.